lib/src/coap_message.dart

```diff
@@ -322,7 +322,7 @@ class CoapMessage {
   }
 
   /// Sets the payload and media type of this CoAP message.
-  void setPayloadMedia(final String? payload, final CoapMediaType mediaType) {
+  void setPayloadMedia(final String? payload, final CoapMediaType? mediaType) {
     if (payload == null) {
```

Comment:
To align this with `payloadMediaRaw` perhaps we should go with `final String payload` (non null) and skip the null check.

For both functions we could also make `mediaType` a named argument perhaps? So you can do `setPayloadMedia('abc')` without explicitly setting the `mediaType` to `null`, or `setPayloadMedia('abc', mediaType: ...)` (which is a bit clearer than using an optional argument).

Reply:
Thanks, good suggestion :) I tried to incorporate it in the latest commit, let me know if you see any potential for improvement :)